extensions/git/src/commands.ts

```diff
@@ -2653,6 +2654,24 @@ export class CommandCenter {
 		await repository.branch(branchName, true, target);
 	}
 
+	private async pickRef<T extends QuickPickItem>(items: Promise<T[]>, placeHolder: string): Promise<T | undefined> {
+		const listeners: Disposable[] = [];
+		const quickPick = window.createQuickPick<T>();
+		quickPick.placeholder = placeHolder;
+		quickPick.sortByLabel = false;
+		quickPick.items = await items;
+		listeners.push(quickPick.onDidHide(() => {
```

Comment:
If the quick input is hidden (user presses ESC) this method will never return. I believe that the `onDidHide` event listener needs to go inside the `choice` method. `onDidHide` should only run `resolve(undefined)`. Then on line 2671, we only need to dispose the quickPick (this will hide + dispose) and the listeners.